### DISCUSSION TITLE
Execute interactive commands with ChildProcess

### DISCUSSION THREAD

**Discussion by jdbravo** [reactphp/reactphp#503] [Q&A]
Can I use ChildProcess to execute comands like bash, or top?

If I try to run top like this:
$process = new Process('top'); the application ends without any output.

And if I try to execute "bash", like this:
$process = new Process('bash'); the application just hangs but without any stdout printed.

Am I missing something?

Thanks,

**Comment by SimonFrings** [comment]
@jdbravo  Interesting question :+1:

> If I try to run top like this:
$process = new Process('top'); the application ends without any output.

The `top` command is designed to run forever in an interactive session by default. These interactive sessions are usually detected by based on TTY or environment variables (depending on program), which can both can be used with ReactPHP. If you want to run top once and then exit, you can take a look at https://unix.stackexchange.com/questions/147471/is-there-a-way-to-get-top-to-run-exactly-once-and-exit.

> And if I try to execute "bash", like this:
$process = new Process('bash'); the application just hangs but without any stdout printed.

The `bash` doesn't hang here, it waits for your input, so if your not sending anything then nothing will happen. You can try to execute something like `echo hello\n`  and should expect to see the "hello" being printed.

Hope this helps :+1:
